### PR TITLE
Improve `CheckoutOrderPlacedEvent` with SalesChannelContext

### PR DESCRIPTION
**Pull request by ghost**
### 1. Why is this change necessary?
- The information about the SalesChannelContext is currently missing in a subscriber on the `CheckoutOrderPlacedEvent`, even though we have the SalesChannelContext where we create the event.

### 2. What does this change do, exactly?
* Changed `Shopware\Core\Checkout\Cart\Event\CheckoutOrderPlacedEvent` by extending the constructor with SalesChannelContext
* Changed `Shopware\Core\Checkout\Cart\Event\CheckoutOrderPlacedEvent` by adding `SalesChannelContextAware` & `CustomerGroupAware`

### 3. Describe each step to reproduce the issue or behaviour.
- Try to access the SalesChannelContext from a subscriber on the `CheckoutOrderPlacedEvent`

### 4. Please link to the relevant issues (if any).
- /

### 5. Checklist

- [X] I have rebased my changes to remove merge conflicts
- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfill them.

### CODE REVIEW

File: src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function store(FlowEventAware $event, array $stored): array
 
         try {
             $stored[CustomerAware::CUSTOMER_ID] = $event->getCustomerId();
-        } catch (CustomerDeletedException) {
+        } catch (CustomerException) {
```
Comment (mitelg) [inline src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php]: in such cases, the exception should stay. the `CustomerDeletedException` should extend `CustomerException` and you can simply return a new instance of `CustomerDeletedException` within the domain exception.

Comment (mitelg) [reply src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php]: othewise, we might catch things, we do not want to catch

File: src/Core/Checkout/Cart/Event/CheckoutOrderPlacedEvent.php [hunk 2]
```diff
@@ -2,84 +2,180 @@
...
-    public function __construct(
-        private readonly Context $context,
-        private readonly OrderEntity $order,
-        private readonly string $salesChannelId,
-        private ?MailRecipientStruct $mailRecipientStruct = null
-    ) {
-    }
-
-    public function getName(): string
+if (Feature::isActive('v6.7.0.0')) {
```
Comment (mitelg) [inline src/Core/Checkout/Cart/Event/CheckoutOrderPlacedEvent.php]: I am not sure if we need that. can't we add a new optional parameter (the salesChannel context) and make it required with 6.7? 

at the same time, deprecating the normal context

Comment (ghost) [reply src/Core/Checkout/Cart/Event/CheckoutOrderPlacedEvent.php]: I have now simply changed the constructor and deprecated the parameters.

File: src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php (hunk 1 above)
Comment (ghost) [reply src/Core/Content/Flow/Dispatching/Storer/CustomerStorer.php]: I changed it back.

File: src/Core/Checkout/Cart/CartException.php [hunk 3]
```diff
@@ -519,6 +524,25 @@ public static function cartMigrationMissingRedisConnection(): self
         );
     }
 
+    /**
+     * @deprecated tag:v6.7.0 - Will be removed, as it is not used anymore in the future
+     */
+    public static function missingSalesChannelContext(): self
+    {
+        Feature::triggerDeprecationOrThrow('v6.7.0.0', \sprintf('Method "%s::%s" will be removed, as it is not used anymore in the future.', self::class, __FUNCTION__));
```
Comment (cyl3x) [inline src/Core/Checkout/Cart/CartException.php]:
```suggestion
        Feature::triggerDeprecationOrThrow('v6.7.0.0', Feature::deprecatedMethodMessage(self::class, __FUNCTION__, 'v6.7.0.0');
```

Comment (ghost) [reply src/Core/Checkout/Cart/CartException.php]: @cyl3x I changed it.

**Comment by cyl3x** [comment]
@akf-bw Thanks for your contribution :tada: :blue_heart:

**Pull request closed without merging**